dds_web/scheduled_tasks.py

```diff
@@ -22,3 +22,48 @@ def change_status_to_expired():
         # ):
 
         #     flask.current_app.logger.debug("Project: %s - Expires: %s", project, project.expires)
+
+
+@scheduler.task("interval", id="expired_to_archived", seconds=30, misfire_grace_time=1)
+def set_expired_to_archived():
+    import sqlalchemy
+    from dds_web import db
+    from dds_web.database import models
+    from dds_web.errors import DatabaseError
+    from dds_web.utils import current_time
+    from dds_web.api.project import ProjectStatus
+
+    with scheduler.app.app_context():
+        archived_projs = db.session.query(models.ProjectStatuses.project_id).filter(
+            models.ProjectStatuses.status == "Archived"
+        )
+        expired_projs = models.ProjectStatuses.query.filter(
+            models.ProjectStatuses.project_id.notin_(archived_projs),
+            models.ProjectStatuses.status == "Expired",
+            models.ProjectStatuses.deadline <= current_time(),
+        ).all()
```

Comment:
I think we need to filter for current_status == "Archived"/"Expired" here, because the projectstatuses table can have all possible statuses for a given project. But then we should query models.Project ...
I am working on it in my PR, but correct me if I am wrong

Reply:
Yes, thats why I only check projects that are not also archived. `current_status` is a property on Project and I found it difficult to query for it.

Reply:
ah right, and you can afford it because Archived cannot go back to anything else 👍